be/src/storage/lake/general_tablet_writer.cpp

```diff
@@ -239,7 +262,14 @@ StatusOr<std::unique_ptr<SegmentWriter>> VerticalGeneralTabletWriter::create_seg
 
 Status VerticalGeneralTabletWriter::flush_columns(std::unique_ptr<SegmentWriter>* segment_writer) {
     uint64_t index_size = 0;
-    RETURN_IF_ERROR((*segment_writer)->finalize_columns(&index_size));
+    if (_segment_writer_finalize_token != nullptr) {
+        return _segment_writer_finalize_token->submit_func([segment_writer, index_size]() mutable {
+            RETURN_IF_ERROR((*segment_writer)->finalize_columns(&index_size));
```

Comment:
The returned error will be ignored, may cause data loss.

Reply:
Should limit the number of tasks that are pending in the token, otherwise it may lead to uncontrolled memory usage.